### PR TITLE
Consolidate kustomize binaries in DSPO Makefile

### PR DESCRIPTION
**Pull request by diegolovison**
## The issue resolved by this Pull Request:
Resolves [RHOAIENG-10819](https://issues.redhat.com//browse/RHOAIENG-10819)

## Description of your changes:
Replace the `kustomize` binary with `$(KUSTOMIZE)`

## Testing instructions
It is automated by `KinD DSPO/DSP Integration Tests / dspo-tests (pull_request)`

## Checklist
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -204,11 +204,7 @@ ENVTEST ?= $(LOCALBIN)/setup-envtest
 ## Tool Versions
 arch:= $(shell uname -m)
 
-ifeq ($(arch), ppc64le)
-KUSTOMIZE_VERSION ?= v5.1.0
-else
-KUSTOMIZE_VERSION ?= v3.8.7
-endif
+KUSTOMIZE_VERSION ?= v5.2.1
```
Comment (hbelmiro) [inline Makefile:207]: Just to confirm, does `v5.2.1` work for other architectures, like Macs?

Comment (diegolovison) [reply Makefile:207]: https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.2.1 supports Linux, Window$, and Mac

cc @rimolive

**Review by hbelmiro — COMMENTED** [review]
/lgtm

**Review by DharmitD — COMMENTED** [review]
DSPO Integration Tests' CI passing.

/lgtm

**Pull request merged**